File: Webpack/src/js/trip.ts
import { Leg } from "../js/leg";

export class Trip {
    public alternative: boolean;
    public cancelled: boolean;
    public valid: boolean;
    public Leg: Leg[];

 constructor(legs: Leg[], alternative?: boolean, valid?: boolean, cancelled?: boolean) {
    this.Leg = legs;

    this.alternative = alternative && alternative || false;
    this.valid = valid && valid || true;
    this.cancelled = cancelled && cancelled || false;
}

}

File: RejseplanAPI/RejseplanAPI/Model/trip.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace RejseplanAPI.Model
{
    public class Trip
    {
        public bool Alternative { get; set; }
        public bool Valid { get; set; }
        public bool Cancelled { get; set; }
        public List<Leg> Legs { get; set; }


        public Trip(List<Leg> legs, bool alternative=false, bool valid=true, bool cancelled=true)
        {
            Alternative = alternative;
            Valid = valid;
            Cancelled = cancelled;
            Legs = legs;
        }
    }
}
